app/Photo.php

```diff
@@ -309,6 +309,20 @@ public function prepareData()
 		return $photo;
 	}
 
+	/**
+	 * Downgrade the quality of the pictures.
+	 *
+	 * @param array $return
+	 */
+	public function downgrade(array &$return)
```

Comment:
Wouldn't it be better to return an empty string as `url` and make sure that the front end handles that correctly? With this change, I think the front end will be unable to tell the dimensions of the image pointed to by `url` (until now, we could use Photo's `width` and `height`). We may not be needing that at the moment, but I think if we implement image zoom, etc, that users have requested, we may need that info...

Reply:
Nicely spotted. Also medium may not even exists...